### PR TITLE
fix(auth0): Changed type of params object and return type of ManagementClient.deleteDeviceCredential

### PR DESCRIPTION
**Pull request by zen-crab**
The `deleteDeviceCredential` method's signature was incorrect. It takes as its params an object with the required field `id`. The return value was changed to `void` as the `deleteDeviceCredential` method does not return a value. The method is a facade for the `/api/v2/device-credentials/{id}` api endpoint, which does not return a payload.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Auth0 Node API](https://auth0.github.io/node-auth0/ManagementClient.html#deleteDeviceCredential) and [Auth0 Management API: ManagementClient.deleteDeviceCredential](https://auth0.com/docs/api/management/v2#!/Device_Credentials/delete_device_credentials_by_id)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/auth0/index.d.ts [hunk 1]
```diff
@@ -1538,8 +1538,8 @@ export class ManagementClient<A = AppMetadata, U = UserMetadata> {
     createDevicePublicKey(data: Data): Promise<User<A, U>>;
     createDevicePublicKey(data: Data, cb: (err: Error, data: any) => void): void;
 
-    deleteDeviceCredential(params: ClientParams): Promise<User<A, U>>;
-    deleteDeviceCredential(params: ClientParams, cb: (err: Error, data: any) => void): void;
+    deleteDeviceCredential(params: ObjectWithId): Promise<void>;
```
Comment (peterblazejewicz) [inline types/auth0/index.d.ts:1541]: what is object with id? if anything, should it be something self-describing? (DeleteParams),
they call it `CredentialParameters`-type IMO:
https://github.com/auth0/node-auth0/blob/08d23024e4051516bd2fbb8c8ab94d4b8ef42c9a/src/management/DeviceCredentialsManager.js#L103

Comment (peterblazejewicz) [reply types/auth0/index.d.ts:1541]: as this is one-prop object, we could just inline this type and avoid naming problems altogether

File: types/auth0/auth0-tests.ts [hunk 2]
```diff
@@ -1596,3 +1596,6 @@ management.getDeviceCredentials({ user_id: 'user_id' }).then(deviceCredentials =
 management.getDeviceCredentials({ user_id: 'user_id' }, (err, deviceCredentials) => {
     deviceCredentials; // $ExpectType DeviceCredential[]
 });
+
+management.deleteDeviceCredential({ id: 'id' }).then(() => {});
+management.deleteDeviceCredential({ id: 'id' }, (err) => {});
```
Comment (peterblazejewicz) [inline types/auth0/auth0-tests.ts:1601]:
```suggestion
management.deleteDeviceCredential({ id: 'id' }, err => {});
```
please run default formatter for anything new added:
https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes (prettier)

File: types/auth0/index.d.ts (hunk 1 above)
Comment (sbmelvin) [reply types/auth0/index.d.ts:1541]: `ObjectWithId` is defined as
```
export interface ObjectWithId {
    id: string;
}
```
I tried to follow the patterns I observed in the existing definitions and `ObjectWithId` is used throughout the definition (92 times in total) whenever a params object only contains an `id` field.

Comment (danursin) [reply types/auth0/index.d.ts:1578]: The named type is more consistent with the rest of the api even if it feels like using an existing type for so many different things. If it needs to diverge for a future change, the inline one would need to change anyway. I'm ok with the named ubiquitous one.

Comment (peterblazejewicz) [reply types/auth0/index.d.ts:1578]: Yes, I;m sorry, I missed `ObjectWithId` in other places.

File: types/auth0/auth0-tests.ts [hunk 3]
```diff
@@ -191,26 +191,60 @@ auth.tokens
...
+        console.log(response);
+    },
+);
+auth.passwordGrant(
+    { username: 'username', password: 'password' },
+    { forwardedFor: '12.34.56.78' },
+    (err, response: auth0.TokenResponse) => {
+        console.log(response);
+    },
+);
```
Comment (peterblazejewicz) [inline types/auth0/auth0-tests.ts:211]:
```suggestion
auth.passwordGrant({ username: 'username', password: 'password' }).then(response => {
    response; // $ExpectType TokenResponse
});
auth.passwordGrant({ username: 'username', password: 'password' }, (err, response) => {
    response; // $ExpectType TokenResponse
});
auth.passwordGrant({ username: 'username', password: 'password' }, { forwardedFor: '12.34.56.78' }).then(response => {
    response; // $ExpectType TokenResponse
});
auth.passwordGrant({ username: 'username', password: 'password' }, { forwardedFor: '12.34.56.78' }, (err, response) => {
    response; // $ExpectType TokenResponse
});
```
please retain format mandated by DT setup (unless there is a specific reason to pick other format, like autogenerated content). With those assertion we will test more real life usage (one should not use specific type, unless being forced to)

File: types/auth0/auth0-tests.ts [hunk 4]
```diff
@@ -191,26 +191,60 @@ auth.tokens
...
 
 // SMS/Email OTP Login
-auth.requestEmailCode({email: 'hi@me.co', authParams: {}}).then((response: any) => { console.log(response); });
-auth.requestEmailCode({email: 'hi@me.co', authParams: {}}, (response: any) => { console.log(response); });
+auth.requestEmailCode({ email: 'hi@me.co', authParams: {} }).then((response: any) => {
+    console.log(response);
+});
+auth.requestEmailCode({ email: 'hi@me.co', authParams: {} }, (response: any) => {
+    console.log(response);
+});
```
Comment (peterblazejewicz) [inline types/auth0/auth0-tests.ts:219]:
```suggestion
auth.requestSMSCode({ phone_number: '+1234567890' }, error => {
    error; // $ExpectType Error
});
auth.requestSMSCode({ phone_number: '+1234567890' }).then(error => {
    error; // $ExpectType any
});
```
and same in other places, please do not introduce just formatting changes. If those changes derive from PR focus, move them into new one

**Comment by sbmelvin** [comment]
Removed all the formatting changes caused by prettier.

**Pull request closed without merging**